app/search/searcher.py

```diff
@@ -0,0 +1,40 @@
+from text_query import TextQuery
+import external_search
+import full_text_search
+import indexing
+
+
+class Searcher(object):
+    """A utility class for searching repositories."""
+
+    def __init__(
+        self, repo, external_search_backends, enable_fulltext_search,
+        max_results):
+        self._repo = repo
+        self._external_search_backends = external_search_backends
+        self._enable_fulltext_search = enable_fulltext_search
+        self._max_results = max_results
+
+    def search(self, query_dict):
```

Comment:
Optional: It can take two params query_name and query_location instead of a dictionary. Up to you though.